### DISCUSSION TITLE
[QUESTION] Why are reboots so sketchy?

### DISCUSSION THREAD

**Discussion by withinboredom** [longhorn/longhorn#5782] [General]
## Question

I'm asking this here because this feels like a bug, but a nebulous one. Whenever there are rolling reboots in the cluster (like yesterday's to deal with the latest kernel security update), things never go well. Here are some of the issues that we run into:

1. Sometimes a node comes up and replicas fail to be created on it and replication just continuously retries on the same node, instead of choosing a different node, flapping between healthy/degraded. Rebooting the node until it works correctly is our current workaround.
2. RWX volumes can hang draining, eventually forcing the reboot anyway (our timeout is 15 mins), causing more issues.
3. Sometimes after a reboot, server load simply runs away (IFF there is an RWX volume on the node) into the thousands. Rebooting until it doesn't do this is our current workaround.

There are probably more issues that I can't think of right now, but reboots never go well and require someone to babysit the cluster for a while because Longhorn won't self-heal on its own.

## Environment

 - Longhorn version: 1.4.1
 - Kubernetes version: 1.25
 - Node config
   - OS type and version: Ububtu 22.04
   - CPU per node: 12-16
   - Memory per node: 64
   - Disk type: nvme - software RAID 1
   - Network bandwidth and latency between the nodes: 1Gb, <1ms
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal

**Comment by c3y1huang** [comment]
Yeah, we know... the team is working on enhancing our regression testing process for resilience scenarios involving node disconnection. This is aiming to identify potential/existing issues around those cases.

https://github.com/longhorn/longhorn/issues/5556

**Reply by withinboredom** [reply]
This is awesome! Thanks for pointing out that issue.
